TypeSystem/Reflection/types.h

```diff
@@ -216,7 +229,7 @@ inline uint64_t ROL64(const TypeID type_id, size_t nbits) {
 }
 
 inline TypeID CalculateTypeID(const ReflectedType_Struct& s) {
-  uint64_t hash = current::CRC32(s.CanonicalName());
+  uint64_t hash = current::CRC32(s.TemplateInnerTypeExpandedName());
```

Comment:
This doing CRC32 over a name to get a hash looks repeated in some other place. Consider pulling out this piece of logic into a reusable function if these uses are the same.